Adapter/ShopwareAdapter/RequestGenerator/Product/ConfiguratorSet/ConfiguratorSetRequestGenerator.php

```diff
@@ -41,15 +47,36 @@ public function generate(Product $product)
 
                 $groups[$propertyName]['options'][$propertyValue]['name'] = $propertyValue;
             }
+
+            if ($plentyConfiguratorType !== 'default' && $plentyConfiguratorType !== $property->getType()) {
+                $plentyConfiguratorType = 'default';
+                continue;
+            }
+
+            $plentyConfiguratorType = $property->getType();
         }
 
         if (empty($groups)) {
             return [];
         }
 
+        $type = (int) $this->config->get('product_configurator_type', 0);
+
+        switch ($plentyConfiguratorType) {
+            case 'box':
+                $type = self::STANDARD;
+                break;
+            case 'dropdown':
+                $type = self::DROP_DOWN;
+                break;
+            case 'image':
+                $type = self::IMAGE;
+                break;
+        }
+
         return [
             'name' => $product->getName(),
-            'type' => (int) $this->config->get('product_configurator_type', 0),
```

Comment:
Wenn die Config Option nicht mehr benötigt wird, bitte auch aus dem Backend entfernen und einen entsprechenden Changelog Eintrag setzen.

Reply:
Die Config dient als Fallback, da wir pro Artikel nur einen Konfigurationstyp hinterlegen können. Wenn eine Artikel mit 2 unterschiedlichen Konfigurationstypen aus Plenty kommt greift diese. 

Alternativ könnte ich den ersten Konfigurationstyp nehmen der reinkommt. Dann gibt es aber bestimmt wieder jede menge Forum Threads :/

Reply:
wenn du die Logik sauber im PlentyMarketsAdapter verpackst ist das für mich dann in Ordnung